core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -85,7 +93,7 @@ public long remaining() {
   public void seek(long pos) throws IOException {
     long offset = mPos - mInitPos;
     if (pos < offset) {
-      setUnderStoreStream(pos);
+      setUnderStoreStream(mInitPos + pos);
```

Comment:
Yeah, do you think that makes sense since the calls are using the relative positions?